store/rootmulti/store.go

```diff
@@ -541,9 +543,28 @@ func (rs *Store) CacheMultiStoreWithVersion(version int64) (types.CacheMultiStor
 			// version does not exist or is pruned, an error should be returned.
 			var err error
 			cacheStore, err = store.(*iavl.Store).GetImmutable(version)
+			// if we got error from loading a module store
+			// we fetch commit info of this version
+			// we use commit info to check if the store existed at this version or not
 			if err != nil {
-				return nil, err
+				if commitInfo == nil {
```

Comment:
`commitInfo` will always be nil.

Reply:
I use `rs.GetCommitInfo(version)` to fetch value for `commitInfo`. I only want to call `rs.GetCommitInfo(version)` once so I add a check to see if  `commitInfo` == nil if false then we don't need to use  `rs.GetCommitInfo(version)` to fetch value for `commitInfo` again.

`commitInfo` will always be nil at first, but then it will be set to be not nil.

Reply:
ohhh duhh I didn't realize this is in a loop!